src/hotspot/cpu/aarch64/macroAssembler_aarch64.hpp

```diff
@@ -1173,6 +1173,7 @@ class MacroAssembler: public Assembler {
 
   // Arithmetics
 
+  // Clobber: rscratch2
```

Comment:
I'm not entirely sure why this has been added as part of this fix. It's ok as a comment but 1) I'm not sure why you do not also note that `addptr` also clobbers `rscratch1` and 2) `cmpptr` clobbers `rscratch1 but not `rscratch2` but it is not clear that this comment only applies to `addptr`. Perhaps a newline between the two methods would make that clearer.

Reply:
Fixed.
As to why the comment change is part of this particular fix, I always put a little flag over a pothole when I see one.